### PR TITLE
upper: tests for config file changes

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @landism, @nicks,

Please review the following commits I made in branch maiamcc/changed-config-file-tests:

fa745a75e5a07b2a4e9d2c551a04392358e2c511 (2018-10-22 12:39:33 -0400)
upper: tests for config file changes

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by maiamcc** [comment]
oops sorry meant to request review from @jazzdan not @nicks

File: internal/engine/upper.go [hunk 1]
```diff
@@ -228,6 +228,7 @@ func (u Upper) maybeStartBuild(ctx context.Context, st *store.Store) {
 				return
 			}
 			ms.PendingFileChanges = mountedChangedFiles
+			fmt.Println("changed files:", mountedChangedFiles)
```
Comment (nicks) [inline internal/engine/upper.go]: :eyes:

File: internal/engine/upper_test.go [hunk 2]
```diff
@@ -33,6 +33,13 @@ import (
 	"github.com/windmilleng/tilt/internal/watch"
 )
 
+const (
+	testTiltfile = `def foobar():
```
Comment (jazzdan) [inline internal/engine/upper_test.go]: maybe a slightly more descriptive name?
```suggestion
	testSimpleTiltfile = `def foobar():
```

Comment (jazzdan) [reply internal/engine/upper_test.go]: I think it'd be cool to have a 'lil library of these: multiple manifests, mounts, runs, etc.

File: internal/engine/upper.go (hunk 1 above)
Comment (maiamcc) [reply internal/engine/upper.go]: 😅

File: internal/engine/upper_test.go (hunk 2 above)
Comment (maiamcc) [reply internal/engine/upper_test.go]: i agree! but also the way we load manifests is inconsistent in this file -- some we `load` from the tiltfile, and some we assemble piecemeal, and urgh.

**Pull request merged**